web3.js/test/transaction.test.ts

```diff
@@ -845,6 +845,79 @@ describe('Transaction', () => {
     expect(expectedTransaction.signatures).to.have.length(1);
   });
 
+  it('partially signed transaction', () => {
```

Comment:
I don't _love_ monolithic tests like this. I know we have a _ton_ of them in the Rust repo, but when I write tests I write a shit ton of them, and they all test exactly one thing. The major benefit to writing tests like that is that when you break one, you know exactly how based on the test that broke. With a test called ‘it partially signed transaction’ you have no idea so you have to go digging into the test itself.

Here's an example of the kind of granularity that accomplishes that.

https://github.com/solana-labs/solana/blob/0b479066b3d92e4d93b8814c201e283519bd6ce5/web3.js/test/connection-subscriptions.test.ts#L683-L741

Reply:
100% agree with this, wasn't a fan of the huge modular test either, but wrote it that way because I had looked at a few other tests in the repo to see how they're generally formatted and got that impression. Glad to hear that isn't on purpose & just rewrote it to be more modular